pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,323 @@
+const apiVersion = "extensions/v1beta1"
+
+const podSecurityAnnotation = "serviceaccount.cluster.cattle.io/pod-security"
+const projectIdAnnotation = "field.cattle.io/projectId"
+
+func Register(context *config.UserContext) {
+	logrus.Infof("registering serviceaccount handler for cluster %v", context.ClusterName)
+
+	m := &manager{
+		clusters: context.Management.Management.Clusters(""),
+		bindings: context.RBAC.RoleBindings(""),
+		policies: context.Extensions.PodSecurityPolicies(""),
+		roles:    context.RBAC.Roles(""),
+
+		clusterLister:   context.Management.Management.Clusters("").Controller().Lister(),
+		templateLister:  context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		policyLister:    context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		bindingLister:   context.RBAC.RoleBindings("").Controller().Lister(),
+		roleLister:      context.RBAC.Roles("").Controller().Lister(),
+		namespaceLister: context.Core.Namespaces("").Controller().Lister(),
+		projectLister:   context.Management.Management.Projects("").Controller().Lister(),
+	}
+
+	/*
+		var mutex = &sync.Mutex{}
+		shouldBindPspt := true
+	*/
+
+	context.Core.ServiceAccounts("").AddHandler(
+		"universalServiceAccountHandler",
+		func(key string, obj *v1.ServiceAccount) error {
+			/*
+				r := func() {
+					mutex.Lock()
+					defer mutex.Unlock()
+					if shouldBindPspt {
+						shouldBindPspt = false
+
+						google11, err := m.clusterLister.Get("", "local")
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+						google11.Spec.DefaultPodSecurityPolicyTemplateName = "pspt1000"
+
+						_, err = m.clusters.Update(google11)
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+					}
+				}
+
+				if true {
+					r()
+				}
+			*/
+			logrus.Info("handle service account")
+
+			if obj == nil {
+				logrus.Infof("no service account provided, exiting")
+				// do nothing
+				return nil
+			}
+
+			namespaceName := obj.Namespace
+
+			logrus.Infof("svc acct name: %v namespace: %v", obj.Name, namespaceName)
+
+			// Check if a PSP exists for this cluster
+			policies, err := m.policyLister.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting policies: %v", err)
+			}
+
+			bindings, err := m.bindingLister.List(namespaceName, labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting role bindings: %v", err)
+			}
+
+			// get PSPT
+			// if no PSPT then get default
+			// if no default then exit
+			namespace, err := m.namespaceLister.Get("", namespaceName)
+
+			if err != nil {
+				return fmt.Errorf("error getting namespaces: %v", err)
+			}
+
+			annotation := namespace.Annotations[projectIdAnnotation]
+
+			if annotation == "" {
+				// no project is associated with namespace so don't do anything
+				logrus.Infof("no project is associated with namespace %v", namespaceName)
```

Comment:
I think this log statement should be bumped down to debug. I think it would be potentially confusing/noisy to an operator monitoring rancher logs